### PR TITLE
Ben/mv5/teleporter

### PR DESCRIPTION
**Pull request by benwoo1110**

### CODE REVIEW

File: src/main/java/org/mvplugins/multiverse/core/api/BlockSafety.java [hunk 1]
```diff
@@ -37,43 +38,74 @@ public interface BlockSafety {
...
+
+    /**
+     * Gets the next safe location around the given location.
+     *
+     * @param location  A {@link Location}.
+     * @param tolerance The tolerance.
+     * @param radius    The radius.
+     * @return A safe {@link Location}.
+     */
+    @Nullable Location getSafeLocation(Location location, int tolerance, int radius);
```
Comment (dumptruckman) [inline src/main/java/org/mvplugins/multiverse/core/api/BlockSafety.java:64]: Could you add some info about what tolerance is in these docs?

File: src/main/java/org/mvplugins/multiverse/core/commands/DeleteCommand.java
Comment (dumptruckman) [inline src/main/java/org/mvplugins/multiverse/core/commands/DeleteCommand.java:1]: How come these commands are in this PR that don't seem related to teleporting? Can we separate into a separate PR maybe?

File: src/main/java/org/mvplugins/multiverse/core/api/SafeTTeleporter.java [hunk 2]
```diff
@@ -14,6 +14,7 @@
 /**
  * Used to safely teleport people.
  */
+@Deprecated
 @Contract
 public interface SafeTTeleporter extends Teleporter {
```
Comment (dumptruckman) [inline src/main/java/org/mvplugins/multiverse/core/api/SafeTTeleporter.java:19]: Why leave it in at all?

File: src/main/java/org/mvplugins/multiverse/core/teleportation/TeleportQueue.java [hunk 3]
```diff
@@ -12,7 +12,7 @@ public class TeleportQueue {
 
     private final Map<String, String> teleportQueue;
 
-    public TeleportQueue() {
+    TeleportQueue() {
         teleportQueue = new HashMap<>();
```
Comment (dumptruckman) [inline src/main/java/org/mvplugins/multiverse/core/teleportation/TeleportQueue.java:16]: Does it matter that we're not using a thread safe Map implementation here?

File: src/main/java/org/mvplugins/multiverse/core/teleportation/AsyncSafetyTeleporter.java [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+import org.jetbrains.annotations.Nullable;
+import org.jvnet.hk2.annotations.Service;
+
+import org.mvplugins.multiverse.core.api.BlockSafety;
+import org.mvplugins.multiverse.core.destination.ParsedDestination;
+import org.mvplugins.multiverse.core.utils.result.Result;
+
+@SuppressWarnings("unchecked")
+@Service
+public class AsyncSafetyTeleporter {
```
Comment (dumptruckman) [inline src/main/java/org/mvplugins/multiverse/core/teleportation/AsyncSafetyTeleporter.java:24]: Is it worth distinguishing this as Async if we don't have an alternative?

File: src/main/java/org/mvplugins/multiverse/core/api/SafeTTeleporter.java (hunk 2 above)
Comment (benwoo1110) [reply src/main/java/org/mvplugins/multiverse/core/api/SafeTTeleporter.java:19]: Bcu of the world manager, now that it’s nuked, it can be removed

File: src/main/java/org/mvplugins/multiverse/core/teleportation/TeleportQueue.java (hunk 3 above)
Comment (benwoo1110) [reply src/main/java/org/mvplugins/multiverse/core/teleportation/TeleportQueue.java:16]: thenAccept for completable future is not on an async thread right?

File: src/main/java/org/mvplugins/multiverse/core/teleportation/AsyncSafetyTeleporter.java (hunk 4 above)
Comment (benwoo1110) [reply src/main/java/org/mvplugins/multiverse/core/teleportation/AsyncSafetyTeleporter.java:24]: Good point

File: src/main/java/org/mvplugins/multiverse/core/commands/DeleteCommand.java
Comment (benwoo1110) [reply src/main/java/org/mvplugins/multiverse/core/commands/DeleteCommand.java:1]: The reason why its in this PR, is bcu it was in the old teleporter class, so I cant just remove it since its useful. But i feel it doesnt belong to the teleporter, since just getting a location.

File: src/main/java/org/mvplugins/multiverse/core/api/BlockSafety.java (hunk 1 above)
Comment (benwoo1110) [reply src/main/java/org/mvplugins/multiverse/core/api/BlockSafety.java:64]: yea sure... this method is just moved, so didnt really check it

File: src/main/java/org/mvplugins/multiverse/core/commands/DeleteCommand.java
Comment (benwoo1110) [reply src/main/java/org/mvplugins/multiverse/core/commands/DeleteCommand.java:1]: Oh also.. bcu everything that reference the old teleporter needs to shift

File: src/test/java/org/mvplugins/multiverse/core/inject/InjectionTest.kt [hunk 5]
```diff
@@ -50,12 +48,6 @@ class InjectionTest : TestWithMockBukkit() {
         assertNotNull(multiverseCore.getService(SimpleLocationManipulation::class.java))
     }
 
-    @Test
-    fun `SafeTTeleporter is available as a service`() {
-        assertNotNull(multiverseCore.getService(SafeTTeleporter::class.java))
-        assertNotNull(multiverseCore.getService(SimpleSafeTTeleporter::class.java))
-    }
```
Comment (dumptruckman) [inline src/test/java/org/mvplugins/multiverse/core/inject/InjectionTest.kt:57]: Can you add a test to replace this for the new service(s)?

**Review by dumptruckman — APPROVED** [review]
